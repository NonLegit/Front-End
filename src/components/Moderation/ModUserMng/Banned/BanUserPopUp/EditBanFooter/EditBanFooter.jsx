import { Box } from '@mui/material';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import { StyledBox } from './styles';
import { RedditButton } from '../../../../styles';
import { UserMngButton } from '../../../styles';

/**
 * EditBan popup footer
 * @component
 * @property  {function} handleClickCloseBan close the popup
 * @property  {function} handleUnban unban user and send data to the backend
 * @property  {function} handleEditBan edit banned user data from the duration of be panned to the reason and the ban message etc
 * @return {React.Component} - EditBan popup foooter
 */

function EditBanFooter(props) {
  const { handleClickCloseBan, handleUnban, handleEditBan } = props;
  return (
    <StyledBox>
      <UserMngButton
        fontSize="15px"
        padding="0px 24px 0px 24px"
        fontWeight="bold"
        disableRipple
        disableFocusRipple
        sx={{ color: 'red' }}
        onClick={handleUnban}
        startIcon={<CheckCircleIcon sx={{ color: 'Red' }} />}
      >
        Unban
      </UserMngButton>
      <Box>
        <RedditButton
          fontSize="15px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="outlined"
          onClick={handleClickCloseBan}
          disableRipple
          disableFocusRipple
        >
          Cancel
        </RedditButton>
        <RedditButton
          sx={{ marginLeft: 1 }}
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="contained"
          disableRipple
          disableFocusRipple
          onClick={handleEditBan}
        >
          Edit
        </RedditButton>
      </Box>
    </StyledBox>
  );
}

export default EditBanFooter;
