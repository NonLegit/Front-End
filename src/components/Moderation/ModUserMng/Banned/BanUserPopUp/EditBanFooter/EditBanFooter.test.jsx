import renderer from 'react-test-renderer';
import EditBanFooter from './EditBanFooter';

test('test snapshot', async () => {
  const tree = renderer.create(<EditBanFooter />).toJSON();
  expect(tree).toMatchSnapshot();
});
