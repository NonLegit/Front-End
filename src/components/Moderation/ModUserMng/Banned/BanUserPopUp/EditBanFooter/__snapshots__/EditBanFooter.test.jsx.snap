// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1ucnzwz"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-49zs3k-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    fontSize="15px"
    fontWeight="bold"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    padding="0px 24px 0px 24px"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-gw46wc-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    Unban
  </button>
  <div
    className="MuiBox-root css-0"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1bny8zm-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize="15px"
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="0px 24px 0px 24px"
      tabIndex={0}
      type="button"
    >
      Cancel
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1dubyk3-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize="14px"
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="0px 24px 0px 24px"
      tabIndex={0}
      type="button"
    >
      Edit
    </button>
  </div>
</div>
`;
